turbinia/workers/analysis/windows_acct_test.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the Windows account analysis task."""
+
+import os
+import unittest
+
+from turbinia import config
+from turbinia.workers.analysis import windows_acct
+
+
+class WindowsAccountAnalysisTaskTest(unittest.TestCase):
+  """Tests for WindowsAccountAnalysisTask Task."""
+
+  TEST_DIR = None
+  RAW_CREDS = [
+      'testuser:1000:aad3b435b51404eeaad3b435b51404ee:9c7ae0f76b24aad74254914c2b191633:::',
+      'testlocaluser:1004:aad3b435b51404eeaad3b435b51404ee:29f98734e7aa3df2454621ff3928d121:::',
+      'badpassword:1005:aad3b435b51404eeaad3b435b51404ee:7a21990fcd3d759941e45c490f143d5f:::'
+  ]
+  EXPECTED_CREDENTIALS = {
+      '$NT$29f98734e7aa3df2454621ff3928d121': 'testlocaluser',
+      '$NT$7a21990fcd3d759941e45c490f143d5f': 'badpassword',
+      '$NT$9c7ae0f76b24aad74254914c2b191633': 'testuser'
+  }
+  REGISTRY_REPORT = """#### **Registry analysis found 2 weak password(s)**
+* **2 weak password(s) found:**
+    * User 'badpassword' with password '12345'
+    * User 'testlocaluser' with password 'google'"""
+
+  def setUp(self):
+    super(WindowsAccountAnalysisTaskTest, self).setUp()
+    filedir = os.path.dirname(os.path.realpath(__file__))
+    self.TEST_DIR = os.path.join(filedir, '..', '..', '..', 'test_data')
+
+  def test_extract_windows_hashes(self):
+    """Tests the extract_windows_hashes method."""
+    config.LoadConfig()
+    task = windows_acct.WindowsAccountAnalysisTask()
+
+    # pylint: disable=protected-access
+    creds, credentials = task._extract_windows_hashes(self.TEST_DIR)
+    self.assertDictEqual(credentials, self.EXPECTED_CREDENTIALS)
+    self.assertCountEqual(creds, self.RAW_CREDS)
+
+  def test_analyse_windows_creds(self):
+    """Tests the analyse_windows_creds method."""
+    config.LoadConfig()
+    task = windows_acct.WindowsAccountAnalysisTask()
+
+    (report, priority, summary) = task._analyse_windows_creds(
```

Comment:
I might have missed this with the other PR too, but I wonder if we should mock out the call to `john` to reduce the test time.  Last I checked, tests took about 30s to run, so this adds a decent amount to that.  Normally that doesn't matter when running through github actions on the PR, but it's nice to keep the test time low when running locally.

Reply:
Can do that, but in testing thhe speed to crack is extremely short, since they're all in the password list.  Plus I've added a timeout flag in the Windows one and the test limits it to 10 seconds, which can be adjusted